hpx/hpx_init.hpp

```diff
@@ -779,39 +789,20 @@ namespace hpx
         std::vector<std::string> const& cfg,
         hpx::runtime_mode mode = hpx::runtime_mode_default);
 
-/// \cond NOINTERNAL
+    /// \cond NOINTERNAL
     namespace resource {
 
         // forward declaration only
         class partitioner;
-    }
-/// \endcond
 
-    /// \brief Main entry point for launching the HPX runtime system.
-    ///
-    /// This is a simplified main entry point, which can be used to set up the
-    /// runtime for an HPX application. All initialization parameters for the
-    /// runtime are taken from the resource partitioner object provided.
-    ///
-    /// \param rp           [in] The resource partitioner object to use for
-    ///                     initializing the runtime.
-    /// \param startup      [in] A function to be executed inside a HPX
-    ///                     thread before \p f is called. If this parameter
-    ///                     is not given no function will be executed.
-    /// \param shutdown     [in] A function to be executed inside an HPX
-    ///                     thread while hpx::finalize is executed. If this
-    ///                     parameter is not given no function will be
-    ///                     executed.
-    ///
-    /// \returns            The function returns the value, which has been
-    ///                     returned from the user supplied function \p f.
-    ///
-    /// \note               The created runtime system instance will be
-    ///                     executed in console or worker mode depending on the
-    ///                     configuration passed in `cfg`.
-    inline int init(resource::partitioner& rp,
-        startup_function_type startup = startup_function_type(),
-        shutdown_function_type shutdown = shutdown_function_type());
+        // Utilities to init the thread_pools of the resource partitioner
+        using rp_callback_type = std::function<void(
+                hpx::resource::partitioner&)>;
+        HPX_EXPORT void set_rp_callback(rp_callback_type f);
```

Comment:
This one we don't need, right?

Reply:
What exactly we don't need ? ^^

Reply:
This was connected to the other comment about passing the callback in the struct instead. We don't need `set_rp_callback` if the callback is passed in the struct.

Reply:
Oh the using, you're right !

Reply:
Ah, not sure if I'm misunderstanding but the `using` can stay and is good to have around. I meant that we don't need to store a static `rp_callback_type` in `hpx_init.cpp` and we don't need the `set_rp_callback` to go with it if the callback is in `init_params` instead.

Reply:
Sorry I didn't see your last comment before, yep sure !